### PR TITLE
Feature/tr 916/support rtl hottext interactions

### PR DESCRIPTION
**Pull request by marpesia**
**Related to:** [TR-916](https://oat-sa.atlassian.net/browse/TR-916)

**Description:**
Add RTL support to Hottext item interaction

**Changes:**

- Add exception on SCSS to the direction change position issues

**How to check:**
 
- Select an RTL language on a Hottext item. Or use the test attached to the ticket and check it is in RTL language.

**Requires:**

- Use the composer.json file attached to the ticket.
- [PR Tao-core](https://github.com/oat-sa/tao-core/pull/3091)

### CODE REVIEW

**Review by jsconan — CHANGES_REQUESTED** [review]
It works well when the main layout is in LTR language.
However, it fails when the layout is in RTL.

image

File: views/scss/qti/_hottext.scss [hunk 1]
```diff
@@ -28,25 +28,40 @@
...
+                        left: 0;
+                    }
+                    &:after {
+                        right: 0;
+                    }
+                    }
+                    }
+
+                    &.error {
+                        color: $error;
```
[the remark this answers is not in the document]
Comment (marpesia) [reply views/scss/qti/_hottext.scss:64]: Great that you see this issue on RTL layout, this is happening with every checkbox, I'm going to fix it for all :)
And fixing too my VSCode extension that breaks my indent on saving.

**Comment by marpesia** [comment]
Please check [Tao-core PR](https://github.com/oat-sa/tao-core/pull/3091) related

**Review by jsconan — COMMENTED** [review]
The issue with the layout is still there.

**Review by jsconan — APPROVED** [review]
Well done!

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by meefox — APPROVED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Comment by marpesia** [comment]
Please check the QA comments on the ticket, there is a corner case

**Comment by marpesia** [comment]
Fixed QA corner case commented on the ticket

**Comment by marpesia** [comment]
Fixed ie11 checkbox issue

**Review by jsconan — APPROVED** [review]
It works well, together with #1814 which is a prerequisite.

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: views/scss/qti/_hottext.scss [hunk 2]
```diff
@@ -14,6 +14,14 @@
         text-decoration: inherit;
     }
 
+    @include rtl {
+        .qti-flow-container {
+            p {
```
Comment (meefox) [inline views/scss/qti/_hottext.scss:19]: does it have any class to hang?
to avoid generic selector

File: views/scss/qti/_hottext.scss [hunk 3]
```diff
@@ -28,25 +36,73 @@
...
-                color : $error;
+            &.error {
+                color: $error;
+            }
+        }
+    }
+}
+
+// IE10+ CSS styles go here
+@media all and (-ms-high-contrast: none),
```
Comment (meefox) [inline views/scss/qti/_hottext.scss:87]: for better readability/understandability/maintainability can we move this hack to mixin?
_qti-base next to @mixin rtl

File: views/scss/qti/_hottext.scss (hunk 2 above)
Comment (marpesia) [reply views/scss/qti/_hottext.scss:19]: no, it doesn't :(

File: views/scss/qti/_hottext.scss (hunk 3 above)
Comment (marpesia) [reply views/scss/qti/_hottext.scss]: done!

**Review by meefox — APPROVED** [review]
#
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Pull request merged**